### PR TITLE
Remove gud package

### PR DESCRIPTION
**Pull request by TrySound**
In this diff I removed gud package which is pure webpackism and requires
`global` to be defined. Rollup cannot handle this out of the box.
Moreover it's one more commonjs package which need to be transpiled.
I inlined commonjsGlobal and added gud like function.

Also tweaked rollup config. Migrated to more trusty terser and
simplified external function.

### CODE REVIEW

**Comment by StringEpsilon** [comment]
Thank you for contributing :)

Can you split the one commit to one commit for the rollup config change and one for the `gud` replacement? 

I'll take a proper look at both changes later.

File: rollup.config.js [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
-function isBareModuleId(id) {
-	if (id.startsWith(".")) {
-		return false;
-	}
-	if (id.startsWith("src")) {
-		return false;
-	}
-	return !id.includes(path.join(process.cwd(), "src"));
+function isExternal(id) {
+  return id.startsWith(".") === false && path.isAbsolute(id) === false;
```
Comment (StringEpsilon) [inline rollup.config.js]: Nitpick: Wrong indentation. 

(I should probably add an .editorconfig file.)

File: src/implementation.ts [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+const commonjsGlobal: any =
+  typeof globalThis !== 'undefined'
+  ? globalThis
+  : typeof window !== 'undefined'
+  ? window
+  : typeof global !== 'undefined'
+  ? global
+  : typeof self !== 'undefined'
+  ? self
+  : {};
```
[the remark this answers is not in the document]
Comment (StringEpsilon) [reply src/implementation.ts]: Thinking about it some more: I'm not a big fan of inlining that polyfill. Users are likely to have the proper polyfill anyway - and the package depends on `@babel/runtime` anyway.

**Comment by StringEpsilon** [comment]
Otherwise LGTM.

File: src/implementation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (TrySound) [reply src/implementation.ts]: Any not existing global access except `typeof variable` gives runtime error. I got it here.
https://github.com/rollup/rollup-plugin-commonjs/blob/master/src/helpers.js#L14

Babel runtime does not require from users any polyfilling. `gud` does require. Yes, webpack does this polyfilling out of the box. Rollup does not. This is really node.js specific thing and should not be in browser library.

File: rollup.config.js (hunk 1 above)
Comment (TrySound) [reply rollup.config.js]: I can rebase if you will setup prettier. By default my editor enforce 2-space indentation.

File: src/implementation.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (StringEpsilon) [reply src/implementation.ts]: Then we can either directly use[ `globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) (it's stage-3 now and supported by most browsers) with instructions on how to poly-fill (either through babel/core-js or with a designated polyfill) or shorten the inline version:

```ts
const commonjsGlobal: any =
  typeof globalThis !== 'undefined' // 'global proper'
  ? globalThis
  : typeof window !== 'undefined' 
  ? window // Browser
  : typeof global !== 'undefined'
  ? global // node.js
```

If the bare object is used, I expect breakage anyway in some scenarios (like if this library is duplicated across multiple loaded bundles). And if it doesn't break in those scenarios, then a `global`-ish construct isn't needed anyway.

I'd still prefer not to inline.

[the remark this answers is not in the document]
Comment (TrySound) [reply src/implementation.ts]: It's not supported by node v8. Polyfills are commonjs modules. Ok, I will remove empty object.

**Pull request closed without merging**